pkg/internal/testabilities/testservices/fixture_bhs.go

```diff
@@ -0,0 +1,113 @@
+package testservices
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"regexp"
+	"testing"
+
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/defs"
+	"github.com/go-softwarelab/common/pkg/to"
+	"github.com/jarcoal/httpmock"
+)
+
+type longestChainTipResponse struct {
+	Height        uint   `json:"height"`
+	Hash          string `json:"hash"`
+	Version       uint32 `json:"version"`
+	MerkleRoot    string `json:"merkleRoot"`
+	Timestamp     uint64 `json:"creationTimestamp"`
+	Bits          uint64 `json:"bits"`
+	Nonce         uint32 `json:"nonce"`
+	PreviousBlock string `json:"prevBlockHash"`
+}
+
+type LongestChainTipOptions func(*longestChainTipResponse)
+
+func WithLongestChainTipHeight(h uint) LongestChainTipOptions {
+	return func(l *longestChainTipResponse) {
+		l.Height = h
+	}
+}
+
+type BHSFixture interface {
+	IsUpAndRunning() BHSFixture
+	WillBeUnreachable() error
+	WillRespondWithInternalFailure()
+	WillRespondWithEmptyLongestTipBlockHeader()
+	OnLongestTipBlockHeaderResponseWith(opts ...LongestChainTipOptions)
+}
+
+type bhsFixture struct {
+	testing.TB
+	transport       *httpmock.MockTransport
+	longestChainTip *longestChainTipResponse
+}
+
+func (b bhsFixture) WillRespondWithEmptyLongestTipBlockHeader() {
+	b.transport.RegisterResponder(
+		http.MethodGet, defs.BHSTestURL+"/chain/tip/longest",
+		httpmock.NewJsonResponderOrPanic(http.StatusOK, longestChainTipResponse{}),
+	)
+}
+
+func (b *bhsFixture) OnLongestTipBlockHeaderResponseWith(opts ...LongestChainTipOptions) {
+	for _, o := range opts {
+		o(b.longestChainTip)
+	}
```

Comment:
```suggestion
	cloned := *b.longestChainTip
	for _, o := range opts {
		o(&cloned)
	}
	b.longestChainTip = &cloned
```
think is safer approach would be to clone the struct to avoid direct mutates ?

Reply:
The idea was to allow customization of the response state using the functional options. Each test should work its own dataset, including an instance of the BHS fixture. Could you provide an example use case for your approach, and explain the benefits compared to the current one? 

@chris-4chain  🤔 ?

Reply:
For current state (how it's used right now), I don't see a reason to make a clone. 
They are tests anyway 🤷